File: src/com/sucy/skill/dynamic/condition/CastLevelCondition.kt
package com.sucy.skill.dynamic.condition

import com.sucy.skill.dynamic.CastContext
import com.sucy.skill.facade.api.entity.Actor

class CastLevelCondition : Condition() {
    override val key = "cast level"

    private var min = 1
    private var max = 999

    override fun initialize() {
        super.initialize()

        min = metadata.getInt("min", min)
        max = metadata.getInt("max", max)
    }

    override fun matches(context: CastContext, target: Actor, recipient: Actor): Boolean {
        return context.level in min..max
    }
}

File: site/pages/skill/condition/CastLevelCondition.js
import React from "react";
import {CONDITION_OPTIONS, Type} from "../commonOptions";
import InputType from "../../../component/input/dynamic/InputType";

const CastLevelCondition = {
    name: 'Cast Level',
    type: Type.CONDITION,
    description: 'Applies child effects if this skill is cast at a matching level',
    children: true,
    metadata: [
        ...CONDITION_OPTIONS,
        {
            key: 'min',
            name: 'Minimum',
            description: 'The minimum level for the skill',
            type: InputType.FORMULA,
            initial: '0'
        },
        {
            key: 'max',
            name: 'Maximum',
            description: 'The maximum level for the skill',
            type: InputType.FORMULA,
            initial: '999'
        }
    ]
};

export default CastLevelCondition